test/runtime/samples/element-source-location/_config.js

```diff
@@ -11,13 +11,15 @@ export default {
 
 		assert.deepEqual(h1.__svelte_meta.loc, {
 			file: path.relative(process.cwd(), path.resolve(__dirname, 'main.svelte')),
+			// TODO: fix this. the h1 is on line 5
 			line: 4,
 			column: 0,
 			char: 53
 		});
 
 		assert.deepEqual(p.__svelte_meta.loc, {
 			file: path.relative(process.cwd(), path.resolve(__dirname, 'Foo.svelte')),
+			// TODO: fix this. the p is on line 2
```

Comment:
I think these tests are correct, as source maps are zero line based. There's actually code in `Element/index.js` where it substracts one from the returned location to account for that. Why do you think these are incorrect?

Reply:
source maps are 0-based, but `locator` is 1-based because it gets printed out to the user and users would expect it to be 1-based:
https://github.com/sveltejs/svelte/blob/9aca05b225a8aec42047092720a0127fed8a12e5/src/compiler/compile/Component.ts#L524

I didn't realize that this `loc` is different than what's in `locator` because I missed the `-1` here:
https://github.com/sveltejs/svelte/blob/b8918faf48d57a525363fb7e3a61a5da5122f050/packages/svelte/src/compiler/compile/render_dom/wrappers/Element/index.js#L590

it's pretty confusing to me to keep track of where things are 0-based and where they're 1-based. I wonder if we can use this opportunity to make it 0-based everywhere and just make it 1-based at the very last second when it's printed to a user